src/OrchardCore/Orchard.Environment.Shell/FileShellSettingsConfigurationProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+
+namespace Orchard.Environment.Shell
+{
+    public class FileShellSettingsConfigurationProvider : IShellSettingsConfigurationProvider
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+
+        public FileShellSettingsConfigurationProvider(IHostingEnvironment hostingEnvironment)
+        {
+            _hostingEnvironment = hostingEnvironment;
+        }
+
+        public int Order => 1000;
+
+        public void AddSource(IConfigurationBuilder builder)
+        {
+            builder.AddJsonFile(Path.Combine(_hostingEnvironment.ContentRootPath, "tenants.json"));
+        }
+
+        public void SaveToSource(string name, IDictionary<string, string> configuration)
```

Comment:
That's weird, tenants can come from multiple sources, but only some will be able to save them.
Then you could have a situation where a tenant is loaded from one configuration, but another one save it and you get the values in different files.